server/memphis_handlers_schemas.go

```diff
@@ -102,38 +101,128 @@ func (sh SchemasHandler) getVersionsCount(schemaId primitive.ObjectID) (int, err
+		return []models.SchemaVersion{}, err
+	}
+
+	return schemaVersions, nil
+}
+
+func (sh SchemasHandler) getUsingStationsByName(schemaName string) ([]models.Station, error) {
+	var stations []models.Station
+	filter := bson.M{"schema_name": schemaName, "is_deleted": false}
+	cursor, err := stationsCollection.Find(context.TODO(), filter)
+
+	if err != nil {
+		return []models.Station{}, err
+	}
+	if err = cursor.All(context.TODO(), &stations); err != nil {
+		return []models.Station{}, err
+	}
+
+	return stations, nil
+}
+
+func (sh SchemasHandler) getExtendedSchemaDetails(schema models.Schema) (models.ExtendedSchemaDetails, error) {
+	schemaVersions, err := sh.getSchemaVersionsBySchemaId(schema.ID)
+
+	if err != nil {
+		return models.ExtendedSchemaDetails{}, err
+	}
+
+	var extedndedSchemaDetails models.ExtendedSchemaDetails
+	stations, err := sh.getUsingStationsByName(schema.Name)
+
+	if err != nil {
 		return models.ExtendedSchemaDetails{}, err
 	}
 
+	var usedStations []string
+	if len(stations) == 0 {
+		usedStations = []string{}
+	}
+	for _, station := range stations {
+		usedStations = append(usedStations, station.Name)
+	}
+
 	tagsHandler := TagsHandler{S: sh.S}
 	tags, err := tagsHandler.GetTagsBySchema(schema.ID)
 	if err != nil {
 		return models.ExtendedSchemaDetails{}, err
 	}
 
-	extedndedSchemaDetails := models.ExtendedSchemaDetails{
-		ID:         schema.ID,
-		SchemaName: schema.Name,
-		Type:       schema.Type,
-		Versions:   schemaVersions,
-		Tags:       tags,
+	extedndedSchemaDetails = models.ExtendedSchemaDetails{
+		ID:           schema.ID,
+		SchemaName:   schema.Name,
+		Type:         schema.Type,
+		Versions:     schemaVersions,
+		UsedStations: usedStations,
+		Tags:         tags,
+	}
+
+	return extedndedSchemaDetails, nil
+}
+
+func (sh SchemasHandler) getExtedndedSchema(schemas []models.ExtendedSchema) ([]models.ExtendedSchema, error) {
+	var extedndedSchemaDetails []models.ExtendedSchema
+	for i, schema := range schemas {
+		stations, err := sh.getUsingStationsByName(schema.Name)
+
+		if err != nil {
+			return []models.ExtendedSchema{}, err
+		}
+
+		var used bool
+		if len(stations) > 0 {
+			used = true
+		} else {
+			used = false
+		}
+
+		tagsHandler := TagsHandler{S: sh.S}
+		tags, err := tagsHandler.GetTagsBySchema(schemas[i].ID)
+		if err != nil {
+			return []models.ExtendedSchema{}, err
+		}
+		schemaUpdated := models.ExtendedSchema{
+			ID:                  schema.ID,
+			Name:                schema.Name,
+			Type:                schema.Type,
+			CreatedByUser:       schema.CreatedByUser,
+			CreationDate:        schema.CreationDate,
+			ActiveVersionNumber: schema.ActiveVersionNumber,
+			Used:                used,
+			Tags:                tags,
+		}
+
+		extedndedSchemaDetails = append(extedndedSchemaDetails, schemaUpdated)
 	}
+
+	return extedndedSchemaDetails, nil
+}
+
+func (sh SchemasHandler) GetSchemaDetailsBySchemaName(schemaName string) (models.ExtendedSchemaDetails, error) {
```

Comment:
I think this can starts without capital letter, check this one and all other functions to make sure